compiler/rustc_transmute/src/lib.rs

```diff
@@ -134,13 +133,8 @@ mod rustc {
             use rustc_middle::ty::ScalarInt;
             use rustc_span::symbol::sym;
 
-            let Ok((ty, cv)) = c.eval(tcx, param_env, DUMMY_SP) else {
-                return Some(Self {
```

Comment:
This was sketchy. If a const was failing to eval, we just returned "whatever". 🤔